zkevm-circuits/src/evm_circuit/execution/error_oog_log.rs

```diff
@@ -0,0 +1,366 @@
+
+        cb.call_context_lookup(
+            false.expr(),
+            None,
+            CallContextFieldTag::RwCounterEndOfReversion,
+            rw_counter_end_of_reversion.expr(),
+        );
+
+        // Go to EndTx only when is_root
+        let is_to_end_tx = cb.next.execution_state_selector([ExecutionState::EndTx]);
+        cb.require_equal(
+            "Go to EndTx only when is_root",
+            cb.curr.state.is_root.expr(),
+            is_to_end_tx,
+        );
+
+        // When it's a root call
+        cb.condition(cb.curr.state.is_root.expr(), |cb| {
+            // Do step state transition
+            cb.require_step_state_transition(StepStateTransition {
+                call_id: Same,
+                rw_counter: Delta(4.expr() + cb.curr.state.reversible_write_counter.expr()),
+                ..StepStateTransition::any()
+            });
+        });
+
+        // When it's an internal call, need to restore caller's state as finishing this
+        // call. Restore caller state to next StepState
+        let restore_context = cb.condition(1.expr() - cb.curr.state.is_root.expr(), |cb| {
+            RestoreContextGadget::construct(
+                cb,
+                0.expr(),
+                0.expr(),
+                0.expr(),
+                0.expr(),
+                0.expr(),
+                0.expr(),
+            )
+        });
+        Self {
+            opcode,
+            memory_address,
+            memory_expansion,
+            insufficient_gas,
+            rw_counter_end_of_reversion,
+            restore_context,
+        }
+    }
+
+    fn assign_exec_step(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        block: &Block<F>,
+        _tx: &Transaction,
+        call: &Call,
+        step: &ExecStep,
+    ) -> Result<(), Error> {
+        let [memory_start, msize] =
+            [step.rw_indices[0], step.rw_indices[1]].map(|idx| block.rws[idx].stack_value());
+
+        let memory_address = self
+            .memory_address
+            .assign(region, offset, memory_start, msize)?;
+
+        // Memory expansion
+        self.memory_expansion
+            .assign(region, offset, step.memory_word_size(), [memory_address])?;
+
+        let opcode = step.opcode.unwrap();
+        self.opcode
+            .assign(region, offset, Value::known(F::from(opcode.as_u64())))?;
+        let topic_count = opcode.postfix().expect("opcode with postfix") as u64;
+        assert!(topic_count <= 4);
+
+        // Gas insufficient check
+        self.insufficient_gas.assign(
+            region,
+            offset,
+            F::from(step.gas_left),
+            F::from(step.gas_cost),
+        )?;
+
+        self.rw_counter_end_of_reversion.assign(
+            region,
+            offset,
+            Value::known(F::from(call.rw_counter_end_of_reversion as u64)),
+        )?;
```

Comment:
If we remove this from the constraints, we should remove it here too.

Reply:
add constraint in [920c06b](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1116/commits/920c06b0a2e727c0fd3a3c38db96eed0530b2c01)